### ISSUE TITLE
Docker containers using Weave network do not get backend ip

### ISSUE THREAD

**Issue by Nossnevs** [containous/traefik#2019]
### Do you want to request a *feature* or report a *bug*?
bug

### What did you do?
[Installed weave](https://www.weave.works/docs/net/latest/install/plugin/plugin-v2/) and added traefik to the weave network. All containers connected to the old overlay network gets an IP. All that are connected to the weave network gets http://:80. If i move containers from the weave network to the overlay network they get the ip.

### What did you expect to see?
It works exactly as with the overlay network.

### What did you see instead?

All backends of services having the weave network have http://:80

### Output of `traefik version`: (_What version of Traefik are you using?_)
Traefik version: 1.3.5

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Docker: 17.06-ce

<details>
<summary>Container on treafik network (docker service inspect)</summary>

```json
[
    {
        "ID": "oy9x5kk59fo7sz3jbqmkydpe7",
        "Version": {
            "Index": 20460
        },
        "CreatedAt": "2017-08-28T13:02:41.046140303Z",
        "UpdatedAt": "2017-08-28T13:02:41.051437608Z",
        "Spec": {
            "Name": "test_0",
            "Labels": {
                "traefik.backend.loadbalancer.method": "drr",
                "traefik.docker.network": "weave",
                "traefik.frontend.entryPoints": "http",
                "traefik.frontend.rule": "Host:test-0.example.com",
                "traefik.port": "80"
            },
            "TaskTemplate": {
                "ContainerSpec": {
                    "Image": "nossnevs/weave_test:latest",
                    "Env": [
                        "SERVICE_NAME=test_0"
                    ],
                    "StopGracePeriod": 10000000000,
                    "DNSConfig": {}
                },
                "Resources": {
                    "Limits": {
                        "MemoryBytes": 512000000
                    },
                    "Reservations": {
                        "MemoryBytes": 100000000
                    }
                },
                "RestartPolicy": {
                    "Condition": "any",
                    "Delay": 5000000000,
                    "MaxAttempts": 0
                },
                "Placement": {
                    "Constraints": [
                        "node.hostname!=node3"
                    ]
                },
                "ForceUpdate": 0,
                "Runtime": "container"
            },
            "Mode": {
                "Replicated": {
                    "Replicas": 2
                }
            },
            "UpdateConfig": {
                "Parallelism": 1,
                "FailureAction": "pause",
                "Monitor": 5000000000,
                "MaxFailureRatio": 0,
                "Order": "stop-first"
            },
            "RollbackConfig": {
                "Parallelism": 1,
                "FailureAction": "pause",
                "Monitor": 5000000000,
                "MaxFailureRatio": 0,
                "Order": "stop-first"
            },
            "Networks": [
                {
                    "Target": "vcl40s719ijj687xzn4mhi6qh"
                }
            ],
            "EndpointSpec": {
                "Mode": "vip"
            }
        },
        "Endpoint": {
            "Spec": {
                "Mode": "vip"
            },
            "VirtualIPs": [
                {
                    "NetworkID": "vcl40s719ijj687xzn4mhi6qh",
                    "Addr": "172.20.0.54/16"
                }
            ]
        }
    }
]
```

</details>

<br/>

<details>
<summary>Container on treafik network (docker inspect) </summary>

```json
[
    {
        "ID": "oy9x5kk59fo7sz3jbqmkydpe7",
        "Version": {
            "Index": 20460
        },
        "CreatedAt": "2017-08-28T13:02:41.046140303Z",
        "UpdatedAt": "2017-08-28T13:02:41.051437608Z",
        "Spec": {
            "Name": "test_0",
            "Labels": {
                "traefik.backend.loadbalancer.method": "drr",
                "traefik.docker.network": "weave",
                "traefik.frontend.entryPoints": "http",
                "traefik.frontend.rule": "Host:test-0.example.com",
                "traefik.port": "80"
            },
            "TaskTemplate": {
                "ContainerSpec": {
                    "Image": "nossnevs/weave_test:latest",
                    "Env": [
                        "SERVICE_NAME=test_0"
                    ],
                    "StopGracePeriod": 10000000000,
                    "DNSConfig": {}
                },
                "Resources": {
                    "Limits": {
                        "MemoryBytes": 512000000
                    },
                    "Reservations": {
                        "MemoryBytes": 100000000
                    }
                },
                "RestartPolicy": {
                    "Condition": "any",
                    "Delay": 5000000000,
                    "MaxAttempts": 0
                },
                "Placement": {
                    "Constraints": [
                        "node.hostname!=node3"
                    ]
                },
                "ForceUpdate": 0,
                "Runtime": "container"
            },
            "Mode": {
                "Replicated": {
                    "Replicas": 2
                }
            },
            "UpdateConfig": {
                "Parallelism": 1,
                "FailureAction": "pause",
                "Monitor": 5000000000,
                "MaxFailureRatio": 0,
                "Order": "stop-first"
            },
            "RollbackConfig": {
                "Parallelism": 1,
                "FailureAction": "pause",
                "Monitor": 5000000000,
                "MaxFailureRatio": 0,
                "Order": "stop-first"
            },
            "Networks": [
                {
                    "Target": "vcl40s719ijj687xzn4mhi6qh"
                }
            ],
            "EndpointSpec": {
                "Mode": "vip"
            }
        },
        "Endpoint": {
            "Spec": {
                "Mode": "vip"
            },
            "VirtualIPs": [
                {
                    "NetworkID": "vcl40s719ijj687xzn4mhi6qh",
                    "Addr": "172.20.0.54/16"
                }
            ]
        }
    }
]
```

</details>

<br/>

<details>
<summary>Container on weave network (docker service inspect) </summary>

```json
[
    {
        "ID": "jwig46qfyhbp2dksfewf94z7e",
        "Version": {
            "Index": 20428
        },
        "CreatedAt": "2017-08-28T13:01:21.446125838Z",
        "UpdatedAt": "2017-08-28T13:01:21.457523131Z",
        "Spec": {
            "Name": "test_0",
            "Labels": {
                "traefik.backend.loadbalancer.method": "drr",
                "traefik.docker.network": "weave",
                "traefik.frontend.entryPoints": "http",
                "traefik.frontend.rule": "Host:test-0.example.com",
                "traefik.port": "80"
            },
            "TaskTemplate": {
                "ContainerSpec": {
                    "Image": "nossnevs/weave_test:latest",
                    "Env": [
                        "SERVICE_NAME=test_0"
                    ],
                    "StopGracePeriod": 10000000000,
                    "DNSConfig": {}
                },
                "Resources": {
                    "Limits": {
                        "MemoryBytes": 512000000
                    },
                    "Reservations": {
                        "MemoryBytes": 100000000
                    }
                },
                "RestartPolicy": {
                    "Condition": "any",
                    "Delay": 5000000000,
                    "MaxAttempts": 0
                },
                "Placement": {
                    "Constraints": [
                        "node.hostname!=node3"
                    ]
                },
                "ForceUpdate": 0,
                "Runtime": "container"
            },
            "Mode": {
                "Replicated": {
                    "Replicas": 2
                }
            },
            "UpdateConfig": {
                "Parallelism": 1,
                "FailureAction": "pause",
                "Monitor": 5000000000,
                "MaxFailureRatio": 0,
                "Order": "stop-first"
            },
            "RollbackConfig": {
                "Parallelism": 1,
                "FailureAction": "pause",
                "Monitor": 5000000000,
                "MaxFailureRatio": 0,
                "Order": "stop-first"
            },
            "Networks": [
                {
                    "Target": "yhmpqxrwd98bnkhyz03ue7xvu"
                }
            ],
            "EndpointSpec": {
                "Mode": "vip"
            }
        },
        "Endpoint": {
            "Spec": {
                "Mode": "vip"
            },
            "VirtualIPs": [
                {
                    "NetworkID": "yhmpqxrwd98bnkhyz03ue7xvu",
                    "Addr": "172.21.0.3/16"
                }
            ]
        }
    }
]
```

</details>

<br/>

<details>
<summary>Container on weave network (docker inspect) </summary>

```json
[
    {
        "ID": "k4rwqu3x6i683s9yf8aqf3a00",
        "Version": {
            "Index": 20477
        },
        "CreatedAt": "2017-08-28T13:05:50.144084629Z",
        "UpdatedAt": "2017-08-28T13:05:50.151646718Z",
        "Spec": {
            "Name": "test_0",
            "Labels": {
                "traefik.backend.loadbalancer.method": "drr",
                "traefik.docker.network": "weave",
                "traefik.frontend.entryPoints": "http",
                "traefik.frontend.rule": "Host:test-0.example.com",
                "traefik.port": "80"
            },
            "TaskTemplate": {
                "ContainerSpec": {
                    "Image": "nossnevs/weave_test:latest",
                    "Env": [
                        "SERVICE_NAME=test_0"
                    ],
                    "StopGracePeriod": 10000000000,
                    "DNSConfig": {}
                },
                "Resources": {
                    "Limits": {
                        "MemoryBytes": 512000000
                    },
                    "Reservations": {
                        "MemoryBytes": 100000000
                    }
                },
                "RestartPolicy": {
                    "Condition": "any",
                    "Delay": 5000000000,
                    "MaxAttempts": 0
                },
                "Placement": {
                    "Constraints": [
                        "node.hostname!=node3"
                    ]
                },
                "ForceUpdate": 0,
                "Runtime": "container"
            },
            "Mode": {
                "Replicated": {
                    "Replicas": 2
                }
            },
            "UpdateConfig": {
                "Parallelism": 1,
                "FailureAction": "pause",
                "Monitor": 5000000000,
                "MaxFailureRatio": 0,
                "Order": "stop-first"
            },
            "RollbackConfig": {
                "Parallelism": 1,
                "FailureAction": "pause",
                "Monitor": 5000000000,
                "MaxFailureRatio": 0,
                "Order": "stop-first"
            },
            "Networks": [
                {
                    "Target": "yhmpqxrwd98bnkhyz03ue7xvu"
                }
            ],
            "EndpointSpec": {
                "Mode": "vip"
            }
        },
        "Endpoint": {
            "Spec": {
                "Mode": "vip"
            },
            "VirtualIPs": [
                {
                    "NetworkID": "yhmpqxrwd98bnkhyz03ue7xvu",
                    "Addr": "172.21.0.3/16"
                }
            ]
        }
    }
]
```

</details>

<br/>

<details>
<summary>docker network inspect traefik</summary>

```json
[
    {
        "Name": "traefik",
        "Id": "vcl40s719ijj687xzn4mhi6qh",
        "Created": "2017-08-28T09:56:36.408669828Z",
        "Scope": "swarm",
        "Driver": "overlay",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.20.0.0/16",
                    "IPRange": "172.20.0.0/16",
                    "Gateway": "172.20.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "478330eeda2bb17d9f82ee818e1fc46109b3a01695dc9e220336ac2e183da450": {
                "Name": "traefik.1mg4ik7kxufy085d3clsu5v8u.9v0puohw448mmupakmn4gz4vt",
                "EndpointID": "b58716f98db63f0664925219580fb105572f31e783e8cddaf2fb4f3228723bf6",
                "MacAddress": "02:42:ac:14:00:46",
                "IPv4Address": "172.20.0.70/16",
                "IPv6Address": ""
            },
        },
        "Options": {
            "com.docker.network.driver.overlay.vxlanid_list": "4097"
        },
        "Labels": {},
        "Peers": [
            {
                "Name": "node1-42cd69130006",
                "IP": "10.135.25.157"
            },
            {
                "Name": "node3-d1f85fa338eb",
                "IP": "10.135.36.196"
            },
            {
                "Name": "node2-fb7e126a5f41",
                "IP": "10.135.8.136"
            },
        ]
    }
]
```

</details>

<br/>

<details>
<summary>docker network inspect weave</summary>

```json
[
    {
        "Name": "weave",
        "Id": "yhmpqxrwd98bnkhyz03ue7xvu",
        "Created": "2017-08-28T09:56:30.448590802Z",
        "Scope": "swarm",
        "Driver": "store/weaveworks/net-plugin:latest_release",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.21.0.0/16",
                    "IPRange": "172.21.0.0/16",
                    "Gateway": "172.21.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "5c88f53cb4a6ac0c58577a00871e4736e87297e4f94ac44fd19eebecd3f22451": {
                "Name": "traefik.ih93vls9lv4u1h11q6t4tg4e5.0urnfc57p2ycxx6pw78uueobh",
                "EndpointID": "3080eb00395100651eb939f0e363f73c726fbcc918b470c965b8fa3e05edb82d",
                "MacAddress": "f6:cf:da:0e:4b:f7",
                "IPv4Address": "172.21.0.5/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {},
        "Peers": [
            {
                "Name": "node1-42cd69130006",
                "IP": "10.135.25.157"
            },
            {
                "Name": "node3-d1f85fa338eb",
                "IP": "10.135.36.196"
            },
            {
                "Name": "node2-fb7e126a5f41",
                "IP": "10.135.8.136"
            }
        ]
    }
]
```

</details>

**Comment by dtomcej** [comment]
@Nossnevs Can you provide the debug log from traefik to show what IP gets parsed for this container setup?

**Comment by Nossnevs** [comment]
```
{
  "backends": {
 
    "backend-test-0": {
      "loadBalancer": {
        "method": "drr"
      }, 
      "servers": {
        "server-test_0-1": {
          "url": "http://:80", 
          "weight": 0
        }, 
        "server-test_0-2": {
          "url": "http://:80", 
          "weight": 0
        }
      }
    },
  "frontends": {
    "frontend-Host-test-0-example-com": {
      "backend": "backend-test-0", 
      "basicAuth": [], 
      "entryPoints": [
        "http"
      ], 
      "passHostHeader": true, 
      "priority": 0, 
      "routes": {
        "route-frontend-Host-test-0-example-com": {
          "rule": "Host:test-0.example.com"
        }
      }
    },
  }
}
```
Also got this

`level=debug msg="Network not found, id: %syhmpqxrwd98bnkhyz03ue7xvu" `

`level=warning msg="Could not find network named 'weave' for container 'test_0.1'! Maybe you're missing the project's prefix in the label? Defaulting to first available network." `



The weave network is available on all nodes and works when using it between containers.

There is no network with id syhmpqxrwd98bnkhyz03ue7xvu. The weave network has id yhmpqxrwd98bnkhyz03ue7xvu.

**Comment by kachkaev** [comment]
I had the same problem in a k8s cluster recently (https://github.com/kubernetes/kubernetes/issues/51014). This is probably weave's issue and traefik is not in charge :–) See https://github.com/weaveworks/weave/issues/2924.

**Comment by Nossnevs** [comment]
If that is the case then i think someone that knows Traefik inner workings should describe in details here how/what they expect from weave net.

**Comment by Nossnevs** [comment]
@dtomcej any info that you or any one else at https://github.com/containous can add to move this forward?

**Comment by onimsha** [comment]
I'm having this issue as well. Below is the log from Traefik.
```
traefik-test_lb.1.y0224nk29chw@host.domain   | time="2017-09-25T03:48:04Z" level=debug msg="Network not found, id: %sttfx7idu1gbhcx82qsgtu03yt"
```
Traefik version: 1.3.8

**Comment by pistolero** [comment]
Try pistolero/traefik:weave_fix

**Comment by onimsha** [comment]
@pistolero  just tested. Now traefik can detect container running with weave net, and route traffic to that container correctly. Thanks !

**Comment by traefiker** [comment]
Closed by #2244.
